pkg/build/linux.go

```diff
@@ -216,6 +223,20 @@ func kernelBin(arch string) string {
 	}
 }
 
+var linuxCompilerRegexp = regexp.MustCompile("#define\\s+LINUX_COMPILER\\s+\"(.*)\"")
+
+func queryLinuxCompiler(kernelDir string) (string, error) {
+	bytes, err := ioutil.ReadFile(filepath.Join(kernelDir, "/include/generated/compile.h"))
+	if err != nil {
+		return "", err
+	}
+	result := linuxCompilerRegexp.FindStringSubmatch(string(bytes))
```

Comment:
Not that it matters much here, but I would use `FindSubmatch` here and then convert the result to string `string(result[1])`. The result is small, the file is large, converting it to string makes a copy and generates garbage.

Reply:
Will take into account, thanks. Fixed.